packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -617,6 +615,12 @@ export function getValidatorApi({
         );
       }
 
+      if (builder.status === "rejected" && engine.status === "rejected") {
+        const failureSource =
+          isEngineEnabled && isBuilderEnabled ? "Builder and engine both" : isEngineEnabled ? "Engine" : "Builder";
```

Comment:
to more naturally read based on the line above (which checks builder first) would swap the order of checks (also matches log message), and for 2nd conditions would check isBuilderEnabled, more common case and feels more correct to default to engine here